tracing-appender/src/inner.rs

```diff
@@ -0,0 +1,96 @@
+use std::io;
+use std::io::Write;
+
+use crate::rolling::{Rotation, WriterFactory};
+use chrono::prelude::*;
+use std::fmt::Debug;
+use std::path::Path;
+
+#[derive(Clone, Debug)]
+pub(crate) struct InnerAppender<F: WriterFactory + Send> {
```

Comment:
As far as I can tell, `F` is _always_ `BufWriterFactory`? I'm not convinced that `InnerAppender` needs to be generic, or that the `WriterFactory` trait is necessary, since there's currently only one implementation. In the future, if we _do_ need to be generic over multiple ways of producing a `Writer`, we can always bring that code back.

Reply:
Yeah, `F` is always `BufWriterFactory`. I'll make this change.